src/gt4py/next/program_processors/runners/dace_fieldview/gtir_dataflow.py

```diff
@@ -83,6 +76,83 @@ class IteratorExpr:
     indices: dict[gtx_common.Dimension, ValueExpr]
 
 
+class DataflowInputEdge(Protocol):
+    """Allows to setup an input edge in a map scope."""
```

Comment:
Maybe the doc string of the class could be a bit more detailed.
I get the concept, like it and I think it is probably the right way to go.
But I would also elaborate how this class is intended to be used, in which situations.
And for completes you should also mention the `connect` function and its relation to the outflow edge.